services/network/HttpClientImpl.hpp

```diff
@@ -138,6 +141,37 @@ namespace services
             bool done = false;
         };
 
+        class SendingStateForwardLimitedSendStream
```

Comment:
Can you consider renaming this class? "LimitedSendStream" points to implementation, not usage. What about SendingStateForwardDefinedSize?